src/dimension.rs

```diff
@@ -10,6 +11,88 @@ pub fn stride_offset(n: Ix, stride: Ix) -> isize
     (n as isize) * ((stride as Ixs) as isize)
 }
 
+/// Check whether `stride` is strictly positive
+#[inline]
+pub fn stride_is_positive(stride: Ix) -> bool
+{
+    (stride as Ixs) > 0
+}
+
+/// Return the axis ordering corresponding to the fastest variation
+///
+/// Assumes that no stride value appears twice. This cannot yield the correct
+/// result the strides are not positive.
+fn fastest_varying_order<D: Dimension>(strides: &D) -> D
+{
+    let mut sorted = strides.clone();
+    sorted.slice_mut().sort();
+    let mut res = strides.clone();
+    for (ind, val) in strides.slice().iter().enumerate() {
+        // cannot binary search in unsorted...
```

Comment:
I'd use `.unwrap()` since it's expected to never be None. No reason to insert a panic with formatting here, that's just awful in the generated code unfortunately.

Compare asm in release mode here: http://is.gd/UFzf2y